base/data_source/datasource_details.go

```diff
@@ -0,0 +1,57 @@
+package data_source
+
+import "strings"
+
+//To handle availablePermissions on Data Source level correctly, we need to import a DataObject of type datasource.
+//DataSourceDetails is a helper to abstract this from the plugin and just expose some setters. The import file creator then automatically prepends the import file with this DataObject
+type DataSourceDetails struct {
+	dataSource DataObject
+}
+
+func newDataSourceDetails(dataSourceId string) DataSourceDetails {
+	return DataSourceDetails{
+		dataSource: DataObject{
+			ExternalId:           dataSourceId,
+			Name:                 dataSourceId,
+			FullName:             dataSourceId,
+			Type:                 "datasource",
+			Description:          "",
+			ParentExternalId:     "",
+			Tags:                 nil,
+			AvailablePermissions: []string{},
+		},
+	}
+}
+
+func (d *DataSourceDetails) SetName(name string) {
+	d.dataSource.Name = name
+}
+
+func (d *DataSourceDetails) SetFullname(name string) {
+	d.dataSource.FullName = name
+}
+
+func (d *DataSourceDetails) SetDescription(desc string) {
+	d.dataSource.Description = desc
+}
+
+func (d *DataSourceDetails) SetAvailablePermission(permission string) {
```

Comment:
Should be 'AddAvailablePermission' then, I suppose

Reply:
Funny enough I had it first as Add but then changed it to be consistent. Maybe Add is better though